docs/design/architecture_3.0/kata_3.0_architecture.md

```diff
@@ -0,0 +1,108 @@
+## Overview
+In cloud-native scenarios, there is an increased demand for container startup speed, resource consumption, stability, and security, areas where the present Kata Containers runtime is challenged relative to other runtimes. To achieve this, we propose a solid, field-tested and secure Rust version of the kata-runtime.
+
+Also, we provide the following designs:
+
+- Turn key solution with builtin Dragonball Sandbox
+- Async io to reduce resource consumption
+- Extensible framework for multiple services, runtimes and hypervisors
+- Lifecycle management for sandbox and container associated resources
+
+### Rationale for choosing Rust
+
+We chose Rust because it is designed as a system language with a focus on efficiency.
+In contrast to Go, Rust makes a variety of design trade-offs in order to obtain
+good execution performance, with innovative techniques that, in contrast to C or
+C++, provide reasonable protection against common memory errors (buffer
+overflow, invalid pointers, range errors), error checking (ensuring errors are
+dealt with), thread safety, ownership of resources, and more.
+
+These benefits were verified in our project when the Kata Containers guest agent
+was rewritten in Rust. We notably saw a significant reduction in memory usage
+with the Rust-based implementation.
+
+
+## Design
+### Architecture
```

Comment:
Thanks, it will be fixed in next version

Reply:
> Some specific questions that I don't think this PR answers:

Good questions! Give some feedbacks here and update the doc to include those answers too:)
 
> 
> * Are the "service", "message dispatcher" and "runtime handler" all part of the single Kata 3.x runtime binary?

Yes, all those modules will be packed into one binary.

> * What is the name of the Kata 3.x runtime binary?

Apparently we can't use `containerd-shim-v2-kata` because it's already used. 
We are facing the hardest issue of "naming" again.  Any suggestions are welcomed:)
Internally we use containerd-shim-v2-rund. 

> * Is the Kata 3.x design compatible with the containerd shimv2 architecture?

Yes, it implements the containerd shim v2 interface/protocol.

> * How will users migrate to the Kata 3.x architecture?
> * I assume Dragonball (as a single process) only works using its own built-in VMM?

Seems we need to define the terms first:)
`runD` is a counterpart of runC, which speaks containerd-shim-v2 and could run a pod/containers.
`Dragonball` is a microvm/VMM, and optimized to run container workload. Sometimes we call it `secure sandbox` instead of microvm/VMM.
So we `Dragonball` is used as a built-in vmm for `runD`, `runD` and `Dragonball` will be built as a single program.
In addition, `runD` supports other VMMs, like qemu, cloud hypervisor and firecracker too, just as kata 2.x.

> * Can the Dragonball system be configured to work using an external Dragonball VMM/hypervisor (aka can it work in the "separate VMM" case)?

Originally it works in that way, and it cost us much efforts to convert to standalone vmm to built-in VMM:)

> * The doc mentions that QEMU, CH and FC support are planned, but it's unclear to me how that would work. Presumably, they would only work in the "separate VMM" case (as happens with Kata 2.x today).

yes, the same model as kata 2.x

> * Please define "upcall" in the documentation.
> * As I think @c3d mentioned, it would be useful to add some details of security boundaries for this (optional) monolithic / "Built-in VMM" case.